### DISCUSSION TITLE
Error when creating new user

### DISCUSSION THREAD

**Discussion by MathieuCruzel** [firefly-iii/firefly-iii#6475] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi,

Upon new user creation, I'm getting this error on a fresh install :

Declaration of FireflyIII\Support\Request\ConvertsDataTypes::integer(string $field): int must be compatible with Illuminate\Http\Request::integer($key, $default = 0)

Both on local and online on self-hosted server with PHP 8.0.24

### Expected behaviour

Creation on new user account

### Debug information

https://flareapp.io/share/17xLjR87#F1

**Comment by JC5** [comment]
Which version of Firefly III?

**Comment by MathieuCruzel** [comment]
Hello, I'm on the last release v5.7.12. I'm not using a docker install btw, just a simple git clone and composer install.

**Comment by JC5** [comment]
I can't replicate it, it works just fine for me. What are your inputs exactly?

**Reply by MathieuCruzel** [reply]
Ok, here are the steps :
- git clone latest release
- edit the .env file
- composer install 
- php artisan the db
- create new user on the app and login
- on the dashboard I try to setup a new account and it fails on creation

**Reply by JC5** [reply]
What amounts do you enter?

**Reply by MathieuCruzel** [reply]
I just insert any number in there, at first I thought it was because I added a comma in the starting balance, but after a fresh isntall I tried everything 1, 10, 1000, 10000, no luck..

**Reply by MathieuCruzel** [reply]
And yeah, it doesn't make sense because the NewUserFormRequest and ConvertsDataTypes has not been touched in 2 years, but I've got the exact same error both on local and online on self-hosted server. I could not find a reason why this happens, and I know a bit about dev, I've been a senior web dev for 10 years so I jsut psoted here on last resort after spending 3 hours going through the code.

**Reply by JC5** [reply]
Very weird. I'll do some more checking over the weekend if that's OK.

**Reply by MathieuCruzel** [reply]
Yes no problem, I'm in no rush, I'm just uping a bug just in case someone else encounters the same problem.

**Comment by JC5** [comment]
Run `composer install` so your packages are up to date, that may help.

**Reply by MathieuCruzel** [reply]
Yes `composer install` is clean and up-to-date, I'm using fr_FR locale if that makes a differences, I tried switching to en_US but no luck there either. I will try a docker install just in case.

**Reply by JC5** [reply]
There may be a possible fix in the `:develop` branch and associated image. Could you check it out?

**Reply by MathieuCruzel** [reply]
Sorry for the delay !

Got a new error, still the same type, some problem with this ConvertDataTypes library : https://flareapp.io/share/W7zLWQbm#F1
This error is both present on local environment (Windows / WAMP stack) and self-hosted (OVH, debian based Plesk LAMP stack)

Got the app working fine on docker though !

**Reply by JC5** [reply]
Thanks! This should be fixed in the next release, 5.8.0.
